server/controllers/notification.js

```diff
@@ -0,0 +1,70 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route PUT /profile/edit
+// @desc edit user profile
+// @access Public
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+const userId = await User.findById(req.user.id);
+const { type, title, description } = req.body;
+const notification = await Notification.create({userId, type, title, description});
+
+  res.status(200).json({
+    success: {
+      notification: notification
+    },
+  });
+
+});
+
+
+exports.markAsRead = asyncHandler(async (req, res, next) => {
+const {notificationId} = req.params;
+const update = { read: true };
+const foundNotification = await Notification.findOneAndUpdate(notificationId, update);
+const readNotification = await foundNotification.save()
+
+  res.status(200).json({
+    success: {
+      notification: readNotification
+    },
+  });
+
+});
+
+exports.retrieveUsersNotifications = asyncHandler(async (req, res, next) => {
+const userId = req.user.id;
+const foundNotifications = await Notification.find({ userId: userId })
+
+    if (!foundNotifications) {
+    res.status(404);
+    throw new Error("No notifications found");
+  }
```

Comment:
There are a couple of issues with this:
- [`model.find`](https://mongoosejs.com/docs/api.html#model_Model.find) will return an array when it suceeds (sometimes empty, sometimes not). And if it fails (i.e. the query rejects), [the `await` keyword will throw an error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description). So in no case will `!foundNotifications` evaluate to true (remember that empty arrays are [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy))!
- If the list of notifications for the user is empty, we shouldn't return a `404`. `404` implies we couldn't locate the resource, which is not true in this case - we located it all fine but it's just empty. So it would be more appropriate to just send the empty collection with a success status code 👍🏼 (you can find more on this [here](https://stackoverflow.com/a/13367198/15220729))

Reply:
Okay so I just deleted these then and will send back the array, empty or not, with the same status code